Python/Product/TestAdapter.Executor/Pytest/PytestExtensions.cs

```diff
@@ -35,47 +36,64 @@ static class PyTestExtensions {
             var sourceParts = sourceAndLineNum.Split(':');
             if (sourceParts.Length != 2 ||
                 !Int32.TryParse(sourceParts[1], out line)) {
-                throw new FormatException(String.Format("Invalid source: {0}",test.ToString()));
+                throw new FormatException(String.Format("Invalid source: {0}", test.ToString()));
             }
 
             return (sourceParts[0], line);
         }
 
-        public static TestCase ToVsTestCase(this PytestTest test, string source, int line, Dictionary<string, PytestParent> parentMap) {
+        public static TestCase ToVsTestCase(
+            this PytestTest test,
+            string source,
+            int line,
+            Dictionary<string, PytestParent> parentMap,
+            string projectHome
+        ) {
             if (parentMap == null) {
                 throw new ArgumentException(nameof(parentMap));
             }
-
             if (String.IsNullOrWhiteSpace(source)) {
                 throw new ArgumentException(nameof(source) + " " + test.ToString());
             }
-            
+            if (String.IsNullOrWhiteSpace(projectHome)) {
+                throw new ArgumentException(nameof(projectHome));
+            }
             if (String.IsNullOrWhiteSpace(test.Name) ||
                 String.IsNullOrWhiteSpace(test.Id)) {
                 throw new FormatException(test.ToString());
             }
-            
-            var fullyQualifiedName = CreateFullyQualifiedTestNameFromId(source, test.Id);
 
+            var pytestId = CreateProperCasedPytestId(source, projectHome, test.Id);
+            var fullyQualifiedName = CreateFullyQualifiedTestNameFromId(source, pytestId);
```

Comment:
Here in discovery phase, we fix any incoming pytestId by rebuilding the relative filename portion